### PR TITLE
cli: parse compatibility argument

### PR DESCRIPTION
**Pull request by vkurchatkin**
Fixes: https://github.com/facebook/prepack/issues/571

### CODE REVIEW

File: src/prepack-cli.js [hunk 1]
```diff
@@ -42,6 +43,23 @@ let flags = {
   delayUnsupportedRequires: false,
   internalDebug: false,
 };
+
+const compatibilityDict: { [string]: ?Compatibility } = {};
+Object.values(CompatibilityValues).map((val: any) => { compatibilityDict[val] = val; })
+
+function parseCompatibility(str: string): Compatibility {
+
+  const compatibility = compatibilityDict[str];
```
Comment (sebmarkbage) [inline src/prepack-cli.js]: This will allow `hasOwnProperty` as a valid compatibility.

File: src/types.js [hunk 2]
```diff
@@ -29,6 +29,11 @@ export type IterationKind = "key+value" | "value" | "key";
 export type SourceType = "module" | "script";
 
 export type Compatibility = "browser" | "jsc-600-1-4-17" | "node-source-maps";
+export const CompatibilityValues = {
+  BROWSER: "browser",
+  MOBILE_JSC: "jsc-600-1-4-17",
+  NODE_SOURCE_MAPS: "node-source-maps",
+};
```
Comment (sebmarkbage) [inline src/types.js]: You should just make this into an array, and then we'll hard code the values in places like `src/realm.js`. No need to take on the runtime dependency on the enum object.

`"jsc-600-1-4-17"` is annoying to type but it'll likely be renamed to something more stable like `"react-native"`.

File: src/prepack-cli.js (hunk 1 above)
Comment (sebmarkbage) [inline src/prepack-cli.js]: If `CompatibilityValues` you could just do `CompatibilityValues.includes(str)` here.

**Comment by sebmarkbage** [comment]
I'll merge this with the hasOwnProperty bug. Feel free to address my comments in a follow up if you want.

Thanks for your help!

File: src/types.js (hunk 2 above)
Comment (vkurchatkin) [inline src/types.js]: OK! I've just notices that `jsc-600-1-4-17` is something that you probably don't want to repeat a lot.

File: src/prepack-cli.js (hunk 1 above)
Comment (vkurchatkin) [inline src/prepack-cli.js]: Was trying to please Flow, but it's definitely not worth it

**Comment by vkurchatkin** [comment]
@sebmarkbage fixed everything

**Pull request merged**